api/paidAction/territoryUnarchive.js

```diff
@@ -34,7 +34,14 @@ export async function perform ({ name, invoiceId, ...data }, { me, cost, tx }) {
   data.userId = me.id
 
   if (sub.userId !== me.id) {
-    await tx.territoryTransfer.create({ data: { subName: name, oldUserId: sub.userId, newUserId: me.id } })
+    const oldUserId = sub.userId
+    const newUserId = me.id
+
+    await tx.territoryTransfer.create({ data: { subName: name, oldUserId, newUserId } })
+
+    // unsubscribe the old user
+    const oldSubscription = await tx.subSubscription.findUnique({ where: { userId_subName: { userId: oldUserId, subName: name } } })
+    if (oldSubscription) await tx.subSubscription.delete({ where: { userId_subName: { subName: name, userId: oldUserId } } })
```

Comment:
do you prefer this or deleteMany?

Reply:
I think `delete` should be fine. But do you need to check first if it exists?

Reply:
yes, delete will error if the record does not exist (eg. if the user unsubscribed manually before transferring the territory) [the workaround is to use deleteMany that will not error](https://github.com/prisma/prisma/discussions/21682).
But in this case, since it is already inside a transaction, maybe this is better?

Reply:
I would use `deleteMany` then